src/Controller/ActionHandler.py

```diff
@@ -0,0 +1,255 @@
+        new_pixmaps = get_pixmaps(pixel_values, window, level)
+
+        self.patient_dict_container.set("window", window)
+        self.patient_dict_container.set("level", level)
+        self.patient_dict_container.set("pixmaps", new_pixmaps)
+
+        self.main_page.update_views()
+
+    def anonymization_handler(self):
+        """
+        Function triggered when the Anonymization button is pressed from the menu.
+        """
+
+        save_reply = QtWidgets.QMessageBox.information(
+            self.main_page.main_window_instance,
+            "Confirmation",
+            "Are you sure you want to perform anonymization?",
+            QtWidgets.QMessageBox.Yes,
+            QtWidgets.QMessageBox.No
+        )
+
+        if save_reply == QtWidgets.QMessageBox.Yes:
+            raw_dvh = self.patient_dict_container.get("raw_dvh")
+            hashed_path = self.main_page.call_class.runAnonymization(raw_dvh)
+            self.patient_dict_container.set("hashed_path", hashed_path)
+            # now that the radiomics data can just get copied across... maybe skip this?
+            radiomics_reply = QtWidgets.QMessageBox.information(
+                self.main_page.main_window_instance,
+                "Confirmation",
+                "Are you sure you want to perform radiomics?",
+                QtWidgets.QMessageBox.Yes,
+                QtWidgets.QMessageBox.No
+            )
+            if radiomics_reply == QtWidgets.QMessageBox.Yes:
+                self.main_page.pyradi_trigger.emit(
+                    self.patient_dict_container.path,
+                    self.patient_dict_container.filepaths,
+                    hashed_path
+                )
+
+    def transect_handler(self):
+        """
+        Function triggered when the Transect button is pressed from the menu.
+        """
+        id = self.main_page.dicom_view.slider.value()
+        dt = self.patient_dict_container.dataset[id]
+        rowS = dt.PixelSpacing[0]
+        colS = dt.PixelSpacing[1]
+        dt.convert_pixel_data()
+        pixmap = self.patient_dict_container.get("pixmaps")[id]
+        self.main_page.call_class.runTransect(
+            self.main_page,
+            self.main_page.dicom_view.view,
+            pixmap,
+            dt._pixel_array.transpose(),
+            rowS,
+            colS
+        )
+
+    def add_on_options_handler(self):
+        self.main_page.add_on_options_controller.show_add_on_options()
+
+    def export_dvh_handler(self):
+        if self.patient_dict_container.has_attribute("raw_dvh"):
+            self.main_page.dvh_tab.export_csv()
+        else:
+            QtWidgets.QMessageBox.information(self.main_page,
+                                              "Unable to export DVH",
+                                              "DVH cannot be exported as there is no DVH present.",
+                                              QtWidgets.QMessageBox.Ok)
+
+    def action_exit_handler(self):
+        sys.exit()
```

Comment:
For a Qt based application, in general one should not call sys.exit().
See http://stackoverflow.com/questions/38283705/ddg#38285497

If the action_exit_handler is connected to the appropriate slot, then maybe this is OK, although one might expect some cleanup activity to be taking place, or at least a comment that says "this will get called on the Qt exit signal, so put cleanup calls in here".
I don't know if we have called in to C/C++ anywhere where some memory management or file I/O needs to get cleaned up before exiting, but that would be the classic scenario